refactor(curl): remove unnecessary CMAKE_PREFIX_PATH

jaudiger opened the pull request:
As shown below, there is no need to expose `CMAKE_PREFIX_PATH` in the `curl` recipe, since there is no CMake files present.

```bash
[container@fd7763c91f9e workspace]$ bb packages/curl
Build finished, completed (no new jobs) in 1.84s
Result: 3ef6623484d67f7cdca559b6261400a85564f8390fa034bc39ebdf56a11db755
Writing output
Wrote output to /tmp/output
[container@fd7763c91f9e workspace]$ find /tmp/output -name "*.cmake"
```

kylewlacy reviewed (commented):
[`FindCURL`](https://cmake.org/cmake/help/latest/module/FindCURL.html#module:FindCURL) is a built-in CMake module, so I _think_ this variable has an effect even though curl itself doesn't have a CMake module. I'm not a CMake expert at all though, so not 100% sure

jaudiger commented:
I did something similar to https://github.com/brioche-dev/brioche-packages/pull/610

I tested it with the following project:

src/main.c:

```c
#include <stdio.h>
#include <curl/curl.h>

int main() {
    CURL *curl;

    curl = curl_easy_init();
    if(curl) {
        curl_easy_cleanup(curl);

        printf("CURL initialized and cleaned up successfully.\n");
    } else {
        fprintf(stderr, "Failed to initialize CURL\n");
    }

    return 0;
}
```

CMakeLists.txt

```cmake
cmake_minimum_required(VERSION 4.0)
project(CmakeTestProject
        VERSION 1.0
        LANGUAGES C)

find_package(CURL REQUIRED)

message(STATUS "CURL_INCLUDE_DIRS: ${CURL_INCLUDE_DIRS}")

add_executable(cmake_test src/main.c)
target_link_libraries(cmake_test PRIVATE CURL::libcurl)

include(GNUInstallDirs)
install(TARGETS cmake_test
        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
```

project.bri

```ts
import * as std from "std";
import { cmakeBuild } from "cmake";
import curl from "curl";
import openssl from "openssl";

export default function(): std.Recipe<std.Directory> {
  return cmakeBuild({
    source: Brioche.glob("src", "CMakeLists.txt"),
    dependencies: [std.toolchain, curl({ minimal: true }), openssl],
    runnable: "bin/cmake_test",
  });
};
``

With and without `CMAKE_PREFIX_PATH`, I obtain the same result:

```bash
[container@b2ec1fbd59c7 workspace]$ bb examples/cmake_test/
59169  │ -- Detecting C compile features - done
       │ -- Found CURL: /home/brioche-runner-aed1d9a565c60630ead83cded476f90d4014f63ef431442581da6d8026028bdb/.local/share/brioche/locals/38b1ff19fd2731d1a1df512d91b3151af9660ab600fd9d916362c2167f59e21f/lib/libcurl.so (found version "8.14.0")
       │ -- Configuring done (2.1s)
       │ -- Generating done (0.0s)
       │ -- Build files have been written to: /home/brioche-runner-aed1d9a565c60630ead83cded476f90d4014f63ef431442581da6d8026028bdb/work
       │ [ 50%] Building C object CMakeFiles/cmake_test.dir/src/main.c.o
       │ [100%] Linking C executable cmake_test
       │ [100%] Built target cmake_test
       │ -- Install configuration: ""
       │ -- Installing: /home/brioche-runner-aed1d9a565c60630ead83cded476f90d4014f63ef431442581da6d8026028bdb/.local/share/brioche/outputs/output-aed1d9a565c60630ead83cded476f90d4014f63ef431442581da6d8026028bdb/bin/cmake_test
 0.07s ✓ Process 59253
 3.66s ✓ Process 59169
Build finished, completed 2 jobs in 7.03s
Result: 9837e0b81404bc0b0d5804eb1648d2b93b8fd876a5e6a8339bc22defedd7c8e4
Writing output
Wrote output to /tmp/output


[container@b2ec1fbd59c7 workspace]$ /tmp/output/bin/cmake_test
CURL initialized and cleaned up successfully.
```

So removing `CMAKE_PREFIX_PATH` does not impact the search of the cURL library by CMake.

kylewlacy reviewed (approved):
Awesome, thanks for testing it too!
